src/appleseed/renderer/kernel/lighting/materialsamplers.cpp

```diff
@@ -219,6 +231,7 @@ bool VolumeSampler::sample(
 
     m_volume.scattering_coefficient(
         m_volume_data, m_volume_ray, m_distance, value.m_volume);
+    value.m_volume *= pdf;
```

Comment:
It is for consistency with `BSDF`. Without it `VolumeSampler::sample` would return just a unit-length direction, multiplied by `scattering_coefficient` which is isotropic.
Why doesn't `Volume::sample` return any `value` like `BSDF::sample`? Because in case of phase functions, value = scattering_coef * pdf always. In most cases it would be a redundancy to return `scattering_coef * pdf` and divide it by `pdf` again for MC sampling.

But I will check this again just in case. Previous time you remark about `if (shading_point.hit()) transmission.set(0.0f);` was very right but I haven't realized it.